server/chat/backend/agent/tools/cloud_exec_tool.py

```
@@ -1254,7 +1272,215 @@ def _run_on_account(conn: dict) -> dict:
+
+    Fans out only when no account_id / project_id / context project is set AND
+    more than one active GCP SA is connected. Single-SA users fall through to
+    the legacy path.
+    """
+    if account_id or project_id or not user_id:
+        return None
+    from utils.cloud.cloud_utils import get_selected_project_id
+    try:
+        if get_selected_project_id():
+            return None
+    except Exception as e:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)